### PR TITLE
fix: allow LIMIT 0 for SELECT queries

### PR DESCRIPTION
**Pull request by ygabuev**
This PR enables the correct queries of form `SELECT ... LIMIT 0`.
Before that, the limit-clause wasn't applied to the query.

Closes https://github.com/uptrace/bun/issues/801

### CODE REVIEW

**Comment by ygabuev** [comment]
The new tests fail for `mssql`, I've checked and they also fail for current master. Given that there were no tests for the `LIMIT` clause, I have a felling that the implementation for `LIMIT` and/or `OFFSET` is broken for `mssql`.

File: query_select.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type SelectQuery struct {
 	group      []schema.QueryWithArgs
 	having     []schema.QueryWithArgs
 	order      []schema.QueryWithArgs
-	limit      int32
+	limit      *int32
```
Comment (vmihailenco) [inline query_select.go:34]: let's use `sql.NullInt32` here

**Comment by vmihailenco** [comment]
@ygabuev this looks good, but I can't merge a PR that breaks the build....

**Comment by CL-Jeremy** [comment]
> The new tests fail for `mssql`, I've checked and they also fail for current master. Given that there were no tests for the `LIMIT` clause, I have a felling that the implementation for `LIMIT` and/or `OFFSET` is broken for `mssql`.
> 
> If this is true, it might be a better idea to fix those in a separate PR.
> 
> UPD: this ticket seems related #811

Hi, I just implemented a fix for that problem in #1009. Could you please test if this workaround fixes your problem?

**Pull request closed without merging**